src/gui/settingsDialog/SettingsDialog.cpp

```diff
@@ -165,37 +167,121 @@ void SettingsDialog::initGui()
 	mTreeWidget->addTopLevelItem(imageGrabber);
 	mTreeWidget->addTopLevelItem(uploader);
 	mTreeWidget->addTopLevelItem(annotator);
-    mTreeWidget->addTopLevelItem(hotkeys);
-    mTreeWidget->addTopLevelItem(actions);
-    mTreeWidget->addTopLevelItem(plugins);
+	mTreeWidget->addTopLevelItem(hotkeys);
+	mTreeWidget->addTopLevelItem(actions);
+	mTreeWidget->addTopLevelItem(plugins);
 	mTreeWidget->setHeaderHidden(true);
-    mTreeWidget->setItemSelected(mNavigatorItems[0], true);
-    mTreeWidget->setFixedWidth(mTreeWidget->minimumSizeHint().width() + mScaledSizeProvider->scaledWidth(100));
-    mTreeWidget->expandAll();
+	mTreeWidget->setItemSelected(mNavigatorItems[0], true);
+	mTreeWidget->setFixedWidth(mTreeWidget->minimumSizeHint().width() + mScaledSizeProvider->scaledWidth(100));
+	mTreeWidget->expandAll();
 
-    auto listAndStackLayout = new QHBoxLayout;
-    listAndStackLayout->addWidget(mTreeWidget);
-    listAndStackLayout->addLayout(mStackedLayout);
+	mSearchSettingsLineEdit->setPlaceholderText(tr("Search Settings..."));
+	mSearchSettingsLineEdit->setFixedWidth(mTreeWidget->width());
 
-    auto mainLayout = new QVBoxLayout();
-    mainLayout->addLayout(listAndStackLayout);
-    mainLayout->addLayout(buttonLayout);
+	auto settingsNavigationLayout = new QVBoxLayout();
+	settingsNavigationLayout->addWidget(mSearchSettingsLineEdit);
+	settingsNavigationLayout->addWidget(mTreeWidget);
 
-    setLayout(mainLayout);
+	auto listAndStackLayout = new QHBoxLayout;
+	listAndStackLayout->addLayout(settingsNavigationLayout);
+	listAndStackLayout->addLayout(mStackedLayout);
+
+	auto mainLayout = new QVBoxLayout();
+	mainLayout->addLayout(listAndStackLayout);
+	mainLayout->addLayout(buttonLayout);
+
+	setLayout(mainLayout);
 }
 
 void SettingsDialog::switchTab()
 {
-    mStackedLayout->setCurrentIndex(mNavigatorItems.indexOf(mTreeWidget->currentItem()));
+	mStackedLayout->setCurrentIndex(mNavigatorItems.indexOf(mTreeWidget->currentItem()));
+}
+
+void SettingsDialog::filterSettings(const QString &filterString)
```

Comment:
We should have some tests for this functionality. Best would be to extract this logic in a separate component with interface, then inject it in this class. After that it should be simple to write tests only for this new component.